frontend/app/(tabs)/chat.tsx

```diff
@@ -48,13 +29,119 @@ const ChatItem = ({ name, lastMessage, avatar, postId }) => {
   );
 };
 
+ //In order to make it presentable, we must filter the raw chats data into the opposite user and an image of the food item
+ interface Preview {
+    userId: string;
+    chatId: string;
+    name: string;
+    lastMessage: string;
+    postImg: string; 
+ }
+
 export default function MyChats() {
-  const renderItem = ({ item }) => (
+  const [previews, setPreviews] = useState<Preview[]>([]);
+  const { user } = useUser();
+  
+  //Business Logic
+  /** Given a user's UID, retrieve all chatIds in the 'users' database */
+  const retrieveListOfChats = async (userUid: string) => {
+    console.log("\n\nRetrieving list of chats\n\n");
+    if (!userUid) {
+      console.error("User ID is null...");
+      return;
+    }
+
+    const userDocRef = doc(db, "users", userUid);
+    const userDocSnap = await getDoc(userDocRef);
+    //Handle the case where there's no user retrieved from db
+    if (!userDocSnap.exists()) {
+      console.error("Fatal error: current user could not be retrieved from the database.");
+      return;
+    }
+
+    const chatIds = userDocSnap.data().chats;
+    // Handle the case where there are no retrievable posts.
+    if (!chatIds || chatIds.length == 0) {
+      setPreviews([]);
+      return;
+    }
+
+    // For each chat, retrieve the data from the "chat" database and return as a chat object.
+    const chatPromises = chatIds.map(async (chatId: string) => {
+      const chatDocRef = doc(db, "chats", chatId);
+      const chatDocSnap = await getDoc(chatDocRef);
+
+      if (chatDocSnap.exists()) {
+        console.log()
+        return chatDocSnap.data() as Chat;
+      } else {
+        return null;
+      }
+    });
+
+    const chatsData = await Promise.all(chatPromises);
+    const validChatsData = chatsData.filter((chat) => chat !== null) as Chat[];
+
+    //Take the raw chats data and convert it into a previews array
```

Comment:
(Non-blocking)
Nit: Seems like you're jumping between a lot of different comment styles. No space after the // here. On line 97, there is a space.

Reply:
yes you're right, tight deadlines are fun